lib/parachain/collation_fetching.go

```diff
@@ -0,0 +1,85 @@
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// CollationFetchingRequest represents a request to retrieve
+// the advertised collation at the specified relay chain block.
+type CollationFetchingRequest struct {
+	// Relay parent we want a collation for
+	RelayParent common.Hash
+
+	// Parachain id of the collation
+	ParaID ParaID
+}
+
+// Encode returns the SCALE encoding of the CollationFetchingRequest
+func (c *CollationFetchingRequest) Encode() ([]byte, error) {
+	return scale.Marshal(*c)
+}
+
+// CollationFetchingResponse represents a response sent by collator
+type CollationFetchingResponse scale.VaryingDataType
+
+// Collation represents a requested collation to be delivered
+type Collation struct {
+	CandidateReceipt CandidateReceipt `scale:"1"`
+	PoV              PoV              `scale:"2"`
+}
+
+// PoV represents a Proof-of-Validity block (PoV block) or a parachain block.
+// It contains the necessary data for the parachain specific state transition logic.
+type PoV []byte
```

Comment:
do we need this as a type? why not keep the `Collation.PoV` as just `[]byte`

Reply:
Not really, changed it.